pytket/pytket/rmbqc/mpattern.py

```diff
@@ -277,22 +279,21 @@ def entangle(g: GraphS) -> Circuit:
             for v in range(len(doneround)):
                 doneround[v] = False
             dlist = []
+            #After every timestep we need to sort the list again to make sure that the vertices with the highest vertex degree are always prioritized.
             for v in vlist:
                 if v in finished_edge_pool.keys():
-                    dlist.append(len(g.neighbors(v)) - finished_edge_pool[v])
+                    dlist.append(len(g.neighbors(v)) - finished_edge_pool[v]) #Remaining vertex degree = original vertex degree minus the number of edges already implemented.
                 else:
                     dlist.append(len(g.neighbors(v)))
                     finished_edge_pool[v] = 0
-            vlist = [x for _, x in sorted(zip(dlist, vlist))]
-            vlist.reverse()
-        # The following code is to be uncommented if we want barriers in the circuit.
-        """
-        active_vertices = []
-        for v in g.vertices():
-            if len(g.neighbors(v))>0:
-                active_vertices.append(v)
-        c.add_barrier(active_vertices, active_vertices)
-        """
+            sorted_vlist = [x for _, x in sorted(zip(dlist, vlist))]
```

Comment:
It would be worth thinking about a different data structure  for dlist and vlist if this sorted_vlist is necessary? Seems inefficient to have to zip two lists together and sort